### PR TITLE
Add extensions live tests

### PR DESCRIPTION
**Pull request by pakrym**
Rename test directories to correct name so they get discovered.

### CODE REVIEW

File: sdk/core/Azure.Core/tests/ArgumentTests.cs [hunk 1]
```diff
@@ -55,11 +55,11 @@ public void NotNullOrEmptyCollectionThrowsOnNull()
             Assert.Throws<ArgumentNullException>(() => Argument.AssertNotNullOrEmpty(value, "value"));
         }
 
-        [TestCaseSource(nameof(GetNotNullOrEmptyCollectionThrowsOnEmptyCollectionData))]
```
Comment (heaths) [inline sdk/core/Azure.Core/tests/ArgumentTests.cs]: Why is this stuff commented out?

File: sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using System.Threading.Tasks;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace Azure.AspNetCore.DataProtection.Blobs.Tests
+{
+    public class DataProtectionBlobFunctionalTest
```
Comment (heaths) [inline sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs]: Nit: class names typically end in "Tests". Might aide in discoverability when using class name filters down the road.

File: sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        private static readonly string StorageAccountName = Environment.GetEnvironmentVariable("BLOB_STORAGE_ACCOUNT_NAME");
+        private static readonly string StorageAccountKey = Environment.GetEnvironmentVariable("BLOB_STORAGE_ACCOUNT_KEY");
+
+        [Test]
+        [Category("Live")]
```
Comment (heaths) [inline sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs]: `[LiveOnly]`?

File: sdk/extensions/Azure.AspNetCore.DataProtection.Keys/tests/DataProtectionKeysFunctionalTest.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+using Azure.Security.KeyVault.Keys;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.AspNetCore.DataProtection.Repositories;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace Azure.AspNetCore.DataProtection.Blobs.Tests
+{
+    public class DataProtectionKeysFunctionalTest
```
Comment (heaths) [inline sdk/extensions/Azure.AspNetCore.DataProtection.Keys/tests/DataProtectionKeysFunctionalTest.cs]: Same nit.

Comment (heaths) [reply sdk/extensions/Azure.AspNetCore.DataProtection.Keys/tests/DataProtectionKeysFunctionalTest.cs]: I also noticed in this case the file name is singular as well.

**Review by heaths — APPROVED** [review]

File: sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs (hunk 3 above)
Comment (pakrym) [reply sdk/extensions/Azure.AspNetCore.DataProtection.Blobs/tests/DataProtectionBlobFunctionalTest.cs]: Can't reference the test framework from this test project for a complicated reason that basically comes down to source references suck.

**Pull request merged**